api/v1alpha1/nodepool_types.go

```diff
@@ -286,6 +287,19 @@ type NodePoolPlatform struct {
 
 	// IBMCloud defines IBMCloud specific settings for components
 	IBMCloud *IBMCloudPlatformSpec `json:"ibmcloud,omitempty"`
+
+	// Kubevirt specifies the configuration used when operating on KubeVirt platform.
+	//
+	// +optional
+	// +immutable
+	Kubevirt *KubevirtNodePoolPlatform `json:"kubevirt,omitempty"`
+}
+
+// KubevirtNodePoolPlatform specifies the configuration of a NodePool when operating
+// on KubeVirt platform.
+type KubevirtNodePoolPlatform struct {
+	// Template Spec contains the VirtualMachineInstance specification.
+	Template *kubevirtv1.VirtualMachineInstanceSpec `json:"template,omitempty"`
```

Comment:
do you want to go ahead and adopt the full VMTemplate as I've outlined here? https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/pull/61 

If we're pretty confident this is the api we're going to move forward with. we could adopt it now on the hypershift side and then not have to deal with mutating the hypershift api in a backwards incompatible way later on.

the change on the hypershift api would look like this.

```
// VirtualMachineTemplateSpec defines the desired state of the kubevirt VM.
type VirtualMachineTemplateSpec struct {
        // +kubebuilder:pruning:PreserveUnknownFields
        // +nullable
        ObjectMeta metav1.ObjectMeta `json:"metadata,omitempty"`
        // VirtualMachineSpec contains the VirtualMachine specification.
        Spec kubevirtv1.VirtualMachineSpec `json:"spec,omitempty" valid:"required"`
}

// KubevirtNodePoolPlatform specifies the configuration of a NodePool when operating
// on KubeVirt platform.
type KubevirtNodePoolPlatform struct {
	// Template Spec contains the VirtualMachine specification.
	VirtualMachineTemplate *VirtualMachineTemplateSpec `json:"virtualMachineTemplate,omitempty"`
}
```

Until the capk PR merges, we'd simply only be using the virtualMachineTemplate.Spec.Template section when translating it to the capk KubeVirtMachine... but once capk adopts the full VMtemplate that would change.

I'm fine either way as long as making backwards incompatible changes to the hypershift NodePool api isn't going to cause us issues

Reply:
Thanks @davidvossel 
I changed it according to last capk change